core/scheduler/service/executor_input_compiler.go

```diff
@@ -95,9 +123,15 @@ func (i InputCompiler) Compile(ctx context.Context, job *scheduler.Job, config s
 		return nil, err
 	}
 
-	jobAttributionLabels := fmt.Sprintf("project=%s,namespace=%s,job=%s", job.Tenant.ProjectName(), job.Tenant.NamespaceName(), job.Name)
-	if jobLables, ok := confs[JobAttributionLabelsKey]; ok {
-		confs[JobAttributionLabelsKey] = jobLables + "," + jobAttributionLabels
+	jobLabelsToAdd := map[string]string{
+		"project":   job.Tenant.ProjectName().String(),
+		"namespace": job.Tenant.NamespaceName().String(),
+		"job_name":  job.Name.String(),
+		"job_id":    job.ID.String(),
+	}
+	jobAttributionLabels := getJobLabelsString(jobLabelsToAdd)
+	if jobLabels, ok := confs[JobAttributionLabelsKey]; ok {
```

Comment:
can `confs[JobAttributionLabelsKey]` return empty string ??

Reply:
needs to be checked, 
but its always safe to add a empty string check , fair point , ill fix this , thanks